src/plugins/data/public/query/query_string/dataset_service/types.ts

```diff
@@ -38,7 +38,7 @@ export interface DatasetTypeConfig {
    * Fetches fields for the dataset.
    * @returns {Promise<DatasetField[]>} A promise that resolves to an array of DatasetFields.
    */
-  fetchFields: (dataset: Dataset) => Promise<DatasetField[]>;
+  fetchFields: (services: IDataPluginServices, dataset: Dataset) => Promise<DatasetField[]>;
```

Comment:
This is a breaking change and plugins like NeoDashboards with fail at runtime if this gets deployed without the required changes in that plugin.

Can we introduce a different method and deprecate this? Though I know this is behind a flag...

Reply:
> Can we introduce a different method and deprecate this? Though I know this is behind a flag...

IMO Adding a new function which does the same would lead to more confusion here. Things are changing everyday on this code piece so not sure how can we keep track of how downstream is using this feature.

Reply:
Agreed we need to be flexible in expecting changes to the signature of experimental methods, so let's make services the second parameter to avoid runtime issues

Reply:
updated - made services optional and second param.